db/crud.go

```diff
@@ -1679,17 +1679,29 @@ func (db *Database) RevDiff(docid string, revids []string) (missing, possible []
 	if strings.HasPrefix(docid, "_design/") && db.user != nil {
 		return // Users can't upload design docs, so ignore them
 	}
-	doc, err := db.GetDocument(docid, DocUnmarshalSync)
-	if err != nil {
-		if !base.IsDocNotFoundError(err) {
-			base.WarnfCtx(db.Ctx, base.KeyAll, "RevDiff(%q) --> %T %v", base.UD(docid), err, err)
-			// If something goes wrong getting the doc, treat it as though it's nonexistent.
+
+	var history RevTree
+
+	if db.UseXattrs() {
+		var doc = sgbucket.BucketDocument{}
+		db.Bucket.GetXattr(docid, KSyncXattrName, &doc.Xattr)
+		doc1, _ := unmarshalDocumentWithXattr(docid, doc.Body, doc.Xattr, doc.Cas, DocUnmarshalSync)
```

Comment:
Using a BucketDocument here only serves to make things harder to follow, I think (below we use 'doc' for *document, and we're not actually interested in doc.Body, or setting doc.cas.

We can just call GetXattr into a new []byte, and pass that and the returned cas to unmarshalDocumentWithXattr.